vic/drivers/classic/src/write_model_state.c

```diff
@@ -293,10 +298,19 @@ write_model_state(all_vars_struct *all_vars,
                            filep->statefile);
                 }
                 else {
-                    fprintf(filep->statefile, " %f", energy[veg][band].T[nidx]);
+                    fprintf(filep->statefile, " "ASCII_STATE_FLOAT_FMT, energy[veg][band].T[nidx]);
                 }
             }
 
+            /* Write foliage temperature */
```

Comment:
I see. I was thinking about a complete binary sample dataset, but specifically to this problem testing binary i/o for the state file is enough. Thanks!